compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -513,9 +512,9 @@ pub struct BoundVarReplacer<'me, 'tcx> {
     // These three maps track the bound variable that were replaced by placeholders. It might be
     // nice to remove these since we already have the `kind` in the placeholder; we really just need
     // the `var` (but we *could* bring that into scope if we were to track them as we pass them).
-    mapped_regions: BTreeMap<ty::PlaceholderRegion, ty::BoundRegion>,
-    mapped_types: BTreeMap<ty::PlaceholderType, ty::BoundTy>,
-    mapped_consts: BTreeMap<ty::PlaceholderConst<'tcx>, ty::BoundVar>,
+    mapped_regions: FxHashMap<ty::PlaceholderRegion, ty::BoundRegion>,
+    mapped_types: FxHashMap<ty::PlaceholderType, ty::BoundTy>,
+    mapped_consts: FxHashMap<ty::PlaceholderConst<'tcx>, ty::BoundVar>,
```

Comment:
Does any code depend on the iteration order?

Reply:
Possibly here?

https://github.com/rust-lang/rust/blob/f1ce0e6a00593493a12e0e3662119786c761f375/compiler/rustc_trait_selection/src/traits/project.rs#L464-L484